src/Operation/FindAndModify.php

```diff
@@ -216,21 +216,14 @@ public function execute(Server $server)
         }
 
         $cursor = $server->executeWriteCommand($this->databaseName, new Command($this->createCommandDocument($server)), $this->createOptions());
-        $result = current($cursor->toArray());
-
-        if ( ! isset($result->value)) {
-            return null;
-        }
-
-        if ( ! is_object($result->value)) {
-            throw new UnexpectedValueException('findAndModify command did not return a "value" document');
```

Comment:
Do we no longer need to throw this exception?

Reply:
`findAndModify` will either return a document or `null` depending on the operation and matched values. We're guarding against a missing `value` property in the return statement, but from what I can see we don't need to guard against non-object values being returned from `findAndModify`.